src/main/java/org/gridsuite/explore/server/services/DirectoryService.java

```diff
@@ -146,14 +151,22 @@ private IDirectoryElementsService getGenericService(String type) {
         return iDirectoryElementsService;
     }
 
-    public List<ElementAttributes> getElementsMetadata(List<UUID> ids) {
-        Map<String, List<ElementAttributes>> elementAttributesListByType = getElementsInfos(ids).stream()
+    public List<ElementAttributes> getElementsMetadata(List<UUID> ids, List<String> elementTypes, List<String> equipmentTypes) {
+        Map<String, List<ElementAttributes>> elementAttributesListByType = getElementsInfos(ids, elementTypes)
+                .stream()
                 .collect(Collectors.groupingBy(ElementAttributes::getType));
         List<ElementAttributes> listOfElements = new ArrayList<>();
         for (Map.Entry<String, List<ElementAttributes>> elementAttribute : elementAttributesListByType.entrySet()) {
             IDirectoryElementsService service = getGenericService(elementAttribute.getKey());
             listOfElements.addAll(service.completeElementAttribute(elementAttribute.getValue()));
         }
+
+        if (equipmentTypes != null && !equipmentTypes.isEmpty() && !listOfElements.isEmpty()) {
+            listOfElements = listOfElements.stream()
+                    .filter(element -> "DIRECTORY".equals(element.getType()) || elementTypes.contains(element.getSpecificMetadata().get("equipmentType")))
```

Comment:
equipmentTypes.contains(...) , instead of elementTypes.contains(...)

Reply:
done